src/main/webapp/src/sessions/gapi.session.ts

```diff
@@ -0,0 +1,217 @@
+        .list({
+          calendarId: 'primary',
+          timeMin: selectedDate,
+          timeMax: selectedDateMax,
+          singleEvents: true,
+          orderBy: 'startTime',
+        })
+        .then(response => {
+          const events = response.result.items;
+          this.tasks.clearTasks();
+          if (events.length > 0) {
+            for (let i = 0; i < events.length; i++) {
+              const event = events[i];
+              if (event.summary.indexOf(gardenName) !== -1) {
+                const eventTime = this.getEventTime(event.start.dateTime);
+                this.tasks.createTaskElement(
+                  eventTime,
+                  event.summary,
+                  event.attendees,
+                  event.description
+                );
+              }
+            }
+          }
+        });
+    });
+  }
+
+  /**
+   * Get the start time of the event otherwise return an All Day string
+   * @param dateTime: string containing date and time
+   */
+  getEventTime(dateTime: string): string {
+    if (dateTime) {
+      const dateTimeSplit = dateTime.split('T');
+      const time = dateTimeSplit[1].split('-', 1);
+      return time[0];
+    } else {
+      return 'All Day';
+    }
+  }
+
+  /**
+   * Inserts a new calendar event into the users' primary calendar
+   * @param title - name of the event (known as summary in the calendar api)
+   * @param startDateTime - start date and time of the event in ISO format
+   * @param endDateTime - end date and time of the event in ISO format
+   * @param timezone - set the timezone for the time
+   * @param participants - members invited to event
+   * @param description - (optional) description of the event
+   */
+  createEvent(
+    title: string,
+    startDateTime: string,
+    endDateTime: string,
+    timezone: string,
+    participants: string[],
+    description?: string
+  ) {
+    if (typeof description === 'undefined') {
+      description = '';
+    }
+
+    // Participants are added to the event in the following format
+    const attendees = [];
+    // Current user
+    attendees.push({email: this.getCurrUserEmail()});
+
+    if (participants) {
+      participants.forEach(participant => {
+        attendees.push({email: participant.trim()});
+      });
+    }
+
+    const event = {
+      summary: title,
+      description: description,
+      start: {
+        dateTime: startDateTime,
+        timeZone: timezone,
+      },
+      end: {
+        dateTime: endDateTime,
+        timeZone: timezone,
+      },
+      attendees: attendees,
+    };
+
+    gapi.client.load('calendar', 'v3', () => {
+      gapi.client.calendar.events
+        .insert({
+          calendarId: 'primary',
+          sendUpdates: 'all',
+          resource: event,
+        })
+        .execute();
+    });
+  }
+
+  /**
+   * Returns the signed in user email
+   * @return - user's email
+   */
+  getCurrUserEmail(): string {
+    if (this.auth2.isSignedIn.get()) {
```

Comment:
is this valid? why are you calling `.get()` on `isSignedIn`?

Reply:
It is valid! I saw this example in which `get()` was being used but then I learned `isSignedIn` was the same thing (and well...shorter) I can change this